src/authn/AuthData.ts

```diff
@@ -13,7 +13,7 @@ export default class AuthData implements authnProto.AuthData {
     timestamp = timestamp || new Date()
     return new AuthData({
       walletAddr: walletAddr,
-      createdNs: timestamp.getTime() * 1000,
+      createdNs: timestamp.getTime() * 1000000,
```

Comment:
1 more fix. The timestamp should be nano not micro. But this gives me this error in tests which I'm not sure how to deal with.
```
● authn › can create a token

    Value is larger than Number.MAX_SAFE_INTEGER

      19 |
      20 |   static fromBytes(bytes: Uint8Array): AuthData {
    > 21 |     const res = authnProto.AuthData.decode(bytes)
         |                                     ^
      22 |     return new AuthData(res)
      23 |   }
      24 |

      at longToNumber (node_modules/@xmtp/proto/ts/message_api/v1/authn.pb.ts:261:11)
      at Object.decode (node_modules/@xmtp/proto/ts/message_api/v1/authn.pb.ts:164:31)
      at Function.fromBytes (src/authn/AuthData.ts:21:37)
      at Token.get authData [as authData] (src/authn/Token.ts:25:33)
      at test/authn/Authn.test.ts:25:18
      at fulfilled (test/authn/Authn.test.ts:5:58)

  ● authn › round trips safely

    Value is larger than Number.MAX_SAFE_INTEGER

      19 |
      20 |   static fromBytes(bytes: Uint8Array): AuthData {
    > 21 |     const res = authnProto.AuthData.decode(bytes)
         |                                     ^
      22 |     return new AuthData(res)
      23 |   }
      24 |

      at longToNumber (node_modules/@xmtp/proto/ts/message_api/v1/authn.pb.ts:261:11)
      at Object.decode (node_modules/@xmtp/proto/ts/message_api/v1/authn.pb.ts:164:31)
      at Function.fromBytes (src/authn/AuthData.ts:21:37)
      at Token.get authData [as authData] (src/authn/Token.ts:25:33)
      at test/authn/Authn.test.ts:48:26
      at fulfilled (test/authn/Authn.test.ts:5:58)
```

Reply:
@neekolas can I let you deal with this one and finish this up? 🙏  No rush, this doesn't block me, I've got my token sample.

Reply:
No problem. I've switched to using a `BigInt` for the intermediate number, which is natively supported in browsers and polyfillable everywhere else.

Reply:
Won't be an issue in RN?

Reply:
Not this way. I've explicitly imported the polyfill, so it'll work out of the box.